packages/contracts-periphery/contracts/universal/op-nft/OptimistInviter.sol

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Semver } from "@eth-optimism/contracts-bedrock/contracts/universal/Semver.sol";
+import { AttestationStation } from "./AttestationStation.sol";
+import {
+    SignatureCheckerUpgradeable
+} from "@openzeppelin/contracts-upgradeable/utils/cryptography/SignatureCheckerUpgradeable.sol";
+import {
+    EIP712Upgradeable
+} from "@openzeppelin/contracts-upgradeable/utils/cryptography/draft-EIP712Upgradeable.sol";
+
+/**
+ * @custom:upgradeable
+ * @title OptimistInviter
+ * @notice OptimistInviter is a contract that issues "optimist.can-invite" and
+ *         "optimist.can-mint-from-invite" attestations. Accounts that have a "optimist.can-invite"
+ *         attestation can issue signatures that allow other accounts to claim an invite. The
+ *         invitee uses a claim and reveal flow to claim the invite to an address of their choosing.
+ */
+contract OptimistInviter is Semver, EIP712Upgradeable {
+    /**
+     * @notice Emitted when an invite is claimed.
+     *
+     * @param issuer Address that issued the signature.
+     * @param claimer Address that claimed the invite.
+     */
+    event InviteClaimed(address indexed issuer, address indexed claimer);
+
+    /**
+     * @notice Version identifier, used for upgrades.
+     */
+    uint8 public constant VERSION = 1;
```

Comment:
this feels bad that we have this and we also have semver.  There must be a better way

Reply:
Agree. James brought this up with me as well. For now we can perhaps just sidestep the matter by using `initializer` (without a version) instead of `reinitalizer`.

Reply:
Updated to use `initializer` instead `reinitializer` and removed the VERSION constant.